docs/source/core_concepts/index.rst

```diff
@@ -0,0 +1,512 @@
+  - There is only one final state
+  - The final state **always** has the **largest** state number
+  - Arcs entering the final state **always** have -1 as the label
+  - Arcs that do not enter the final state cannot have -1 as the label
+
+.. HINT::
+
+  Different from other frameworks, FSAs in k2 have only a single final state.
+
+  If you want to convert an FSA from another framework to k2 that contains
+  multiple final states, you can create an extra state and consider it as
+  the super final state. For each final state in the FSA, add an arc to this
+  super final state with label -1 and score 0. The resulting FSA will contain
+  only a single final state.
+
+  Similarly, if it contains multiple start states, you can add a super start
+  state and set both the label and score of the arcs added from the super start
+  state to the start state to 0.
+
+Attributes
+----------
+
+Arbitrary attributes can be attached to the arcs of an FSA.
+For example, we can attach a tensor attribute to an FSA indicating
+the output label of arcs so that the FSA is converted to an FST.
+
+The attached attributes are **automaticaly propagated** through operations,
+with ``autograd`` if they are real-valued tensors.
+
+The following code converts the above simple acceptor to a transducer:
+
+.. code-block:: python
+
+    import k2
+    import torch
+    s = '''
+    0 1 10 0.1
+    1 2 -1 0.2
+    2
+    '''
+    fsa = k2.Fsa.from_str(s)
+    fsa.aux_labels = torch.tensor([100, -1], dtype=torch.int32)
+    fsa.draw('simple_fst.svg')
+
+
+The resulting FST is visualized in :numref:`a simple fst`.
+
+.. _a simple fst:
+.. figure:: images/simple_fst.svg
+    :alt: A simple FST in k2
+    :align: center
+    :figwidth: 600px
+
+    A simple FST in k2.
+
+.. CAUTION::
+
+  There are NO **output labels** in k2. Every arc has a label and you
+  can attach arbitrary attributes with arbitrary name to it.
+
+  If the attached attribute is an N-D tensor, its ``shape[0]`` has to
+  equal the number of arcs in the FSA.
+
+.. NOTE::
+
+  The visualization code handles the attributes ``aux_labels`` specially.
+  Other than this, ``aux_labels`` is like any other attributes attached
+  to the FSA.
+
+Semirings
+---------
+
+k2 supports two kinds of semirings:
+
+  - tropical semiring
+  - log semiring
+
+Tropical semiring
+~~~~~~~~~~~~~~~~~
+
+In tropical semirings, it takes the **max** score of alternative paths.
+
+For example, for the FSA in :numref:`tropical`:
+
+.. _tropical:
+.. figure:: images/fsa2.svg
+    :alt: An FSA with two alternative paths
+    :align: center
+    :figwidth: 600px
+
+    An FSA with two alternative paths to the final states.
+
+There are two paths from the start state to the final state:
+
+  - Path 0: state 0 -> state 1 -> state 3, with score: 0.1 + 0 = 0.1
+  - Path 1: state 0 -> state 2 -> state 3, with score: 0.2 + 0 = 0.2
+
+So in the tropical semiring, the best score is ``max(0.1, 0.2) == 0.2``.
```

Comment:
So in the tropical semiring, we would consider that "total score" of this FSA is ``max(0.1, 0.2) == 0.2``.

Reply:
Fixed.